### PR TITLE
Some fixes for running tests in main

### PR DESCRIPTION
**Pull request by alanking**
Confirmed that `--run_specific_tests` still works as expected and passing nothing still runs all of the tests.

### CODE REVIEW

**Comment by alanking** [comment]
May try to run things through 2to3, but this at least gets things off the ground

**Comment by trel** [comment]
yeps - please # when green.

File: unit_tests/cmake/test_config/irods_client_server_negotiation.cmake [hunk 1]
```diff
@@ -4,7 +4,8 @@ set(IRODS_TEST_SOURCE_FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/main.cpp
                             ${CMAKE_CURRENT_SOURCE_DIR}/src/test_client_server_negotiation.cpp)
 
 set(IRODS_TEST_INCLUDE_PATH ${IRODS_EXTERNALS_FULLPATH_BOOST}/include
-                            ${IRODS_EXTERNALS_FULLPATH_FMT}/include)
+                            ${IRODS_EXTERNALS_FULLPATH_FMT}/include
+                            ${IRODS_EXTERNALS_FULLPATH_JSON}/include)
```
Comment (korydraughn) [inline unit_tests/cmake/test_config/irods_client_server_negotiation.cmake:8]: To keep inline with what @SwooshyCueb has been doing, you're going to want to replace the JSON line with `nlohmann_json::nlohmann_json`.

Comment (alanking) [reply unit_tests/cmake/test_config/irods_client_server_negotiation.cmake:8]: Done

Comment (trel) [reply unit_tests/cmake/test_config/irods_client_server_negotiation.cmake:8]: i see your 'Done', but ... it isn't?

Comment (alanking) [reply unit_tests/cmake/test_config/irods_client_server_negotiation.cmake:8]: Sorry, haven't pushed it yet

**Comment by alanking** [comment]
Awaiting green. Other fixes for tests are now in a separate branch and will come in a later PR one I can confirm everything is passing. Any other comments are welcome, of course

**Comment by trel** [comment]
good here.

**Comment by alanking** [comment]
#'d

**Pull request merged**